compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -403,6 +410,25 @@ fn link_rlib<'a>(
         ab.add_file(&trailing_metadata);
     }
 
+    // Add all bundled static native library dependencies.
+    // Archives added to the end of .rlib archive, see comment above for the reason.
+    if flavor == RlibFlavor::Normal && sess.opts.unstable_opts.packed_bundled_libs {
+        for lib in codegen_results.crate_info.used_libraries.iter() {
```

Comment:
Could you merge this with the other loop over `used_libraries` and put the if's after the match?

Reply:
They cannot be trivially merged because this loop adds entries that should be at the archive end, and the loop above adds entries that should *not* be at the archive end (object files).
The `lib.path`s can be stashed to a vector in the other loop and still added at the archive end here though.